Deployment Guides/Provisioning-Islandora-on-Ubuntu-16.md

```diff
@@ -0,0 +1,1788 @@
+
+sed -i "s|log.application.FileAppender.pathname = /path/to/logs/application.log|log.application.FileAppender.pathname = /srv/cantaloupe/log/application.log|" cantaloupe.properties
+
+sed -i "s|log.application.RollingFileAppender.enabled = false|log.application.RollingFileAppender.enabled = true|" cantaloupe.properties
+
+sed -i "s|log.application.RollingFileAppender.pathname = /path/to/logs/application.log|log.application.RollingFileAppender.pathname = /srv/cantaloupe/log/application.log|" cantaloupe.properties
+
+sed -i "s|log.application.RollingFileAppender.TimeBasedRollingPolicy.filename_pattern = /path/to/logs/application-%d{yyyy-MM-dd}.log|log.application.RollingFileAppender.TimeBasedRollingPolicy.filename_pattern = /srv/cantaloupe/log/application-%d{yyyy-MM-dd}.log|" cantaloupe.properties
+
+sed -i "s|log.error.FileAppender.enabled = false|log.error.FileAppender.enabled = true|" cantaloupe.properties
+
+sed -i "s|log.error.FileAppender.pathname = /path/to/logs/error.log|log.error.FileAppender.pathname = /srv/cantaloupe/log/error.log|" cantaloupe.properties
+
+sed -i "s|log.error.RollingFileAppender.pathname = /path/to/logs/error.log|log.error.RollingFileAppender.pathname = /srv/cantaloupe/log/error.log|" cantaloupe.properties
+
+sed -i "s|log.error.RollingFileAppender.TimeBasedRollingPolicy.filename_pattern = /path/to/logs/error-%d{yyyy-MM-dd}.log|log.error.RollingFileAppender.TimeBasedRollingPolicy.filename_pattern = /srv/cantaloupe/log/error-%d{yyyy-MM-dd}.log|" cantaloupe.properties
+
+sed -i "s|log.access.FileAppender.pathname = /path/to/logs/access.log|log.access.FileAppender.pathname = /srv/cantaloupe/log/access.log|" cantaloupe.properties
+
+sed -i "s|log.access.RollingFileAppender.pathname = /path/to/logs/access.log|log.access.RollingFileAppender.pathname = /srv/cantaloupe/log/access.log|" cantaloupe.properties
+
+sed -i "s|log.access.RollingFileAppender.TimeBasedRollingPolicy.filename_pattern = /path/to/logs/access-%d{yyyy-MM-dd}.log|log.access.RollingFileAppender.TimeBasedRollingPolicy.filename_pattern = /srv/cantaloupe/log/access-%d{yyyy-MM-dd}.log|" cantaloupe.properties
+
+sed -i "s|processor.jp2 = KakaduProcessor|processor.jp2 = OpenJpegProcessor|" cantaloupe.properties
+
+sed -i "s|StreamProcessor.retrieval_strategy = StreamStrategy|StreamProcessor.retrieval_strategy = CacheStrategy|" cantaloupe.properties
+
+sed -i "s|cache.server.source.enabled = false|cache.server.source.enabled = true|" cantaloupe.properties
+
+sed -i "s|cache.server.derivative =*|cache.server.derivative = FilesystemCache|" cantaloupe.properties
+
+sed -i "s|cache.server.worker.enabled = false|cache.server.worker.enabled = true|" cantaloupe.properties
+
+sed -i "s|resolver.static = FilesystemResolver|resolver.static = HttpResolver|" cantaloupe.properties
+
+sed -i "s|HttpResolver.trust_all_certs = false|HttpResolver.trust_all_certs = true|" cantaloupe.properties
```

Comment:
Not sure if we should leave "trust_all_certs" by default. It would be better to get Java/Tomcat to trust the SSL cert.

Reply:
I think for a back end server like this, that doesn't connect to the outside world
it's probably fine. Do we even use ssl? 'https' is disabled.

Reply:
When Adore-Djatoka or Cantaloupe have to retrieve an image from the frontend webserver and it (the webserver) is using HTTPS it will attempt to check the SSL cert. If it doesn't know how to trust the cert the request will fail. Generally, if this is the case you should make sure the CA root certificate/chain is added to the Truststore. Cantaloupe provides an option so it will trust the cert anyway even if it doesn't know how to trust the cert or if it is invalid. Generally, we should avoid doing this in production but it is useful for dev/testing/troubleshooting.

Reply:
I'm going to be in Victoria until the 18th so I won't be able to look at this until then.
I guess we should add something about how to update the Tomcat trust store.
I'm running Cantaloupe stand alone, so that's a different kettle of fish...